allennlp/common/cached_transformers.py

```diff
@@ -43,13 +46,26 @@ def get(
         with `torch.save()`.
     override_weights_strip_prefix : `str`, optional (default = `None`)
         If set, strip the given prefix from the state dict when loading it.
+    reinit_modules: `Optional[Union[int, Tuple[int, ...], Tuple[str, ...]]]`, optional (default = `None`)
+        If this is an integer, the last `reinit_modules` layers of the transformer will be
+        re-initialized. If this is a tuple of integers, the layers indexed by `reinit_modules` will
+        be re-initialized. If this is a tuple of strings, they will be treated as regexes and any
+        module with a name matching the regex will be re-initialized. Re-initializing the last few
+        layers of a pretrained transformer can reduce the instability of fine-tuning on small
+        datasets and may improve performance (https://arxiv.org/abs/2006.05987v3). Has no effect
+        if `load_weights` is `False` or `override_weights_file` is not None.
```

Comment:
I think it would be helpful to have a note here about how we can't guarantee using the `int` or `list[int]` form will work. And if that fails, the user should use the `list[str]` form.

Reply:
Good call! Updated this docstring in `PretrainedTransformerEmbedder` and `cached_transformers.get()` with a note warning that layer indices could fail